src/tests.rs

```diff
@@ -192,6 +192,60 @@ fn encode_into_nonempty_buffer_doesnt_clobber_existing_contents() {
 }
 
 #[test]
+fn encode_config_slice_into_nonempty_buffer_doesnt_clobber_suffix() {
+    let mut orig_data = Vec::new();
+    let mut encoded_data = Vec::new();
+    let mut encoded_data_original_state = Vec::new();
+    let mut decoded = Vec::new();
+
+    let input_len_range = Range::new(0, 1000);
+    let line_len_range = Range::new(1, 1000);
+
+    let mut rng = rand::weak_rng();
+
+    for _ in 0..10_000 {
+        orig_data.clear();
+        encoded_data.clear();
+        encoded_data_original_state.clear();
+        decoded.clear();
+
+        let input_len = input_len_range.ind_sample(&mut rng);
+
+        for _ in 0..input_len {
+            orig_data.push(rng.gen());
+        }
+
+        // plenty of existing garbage in the encoded buffer
+        for _ in 0..10 * input_len {
+            encoded_data.push(rng.gen());
+        }
+
+        encoded_data_original_state.extend_from_slice(&encoded_data);
+
+        let config = random_config(&mut rng, &line_len_range);
+
+        let encoded_size = encoded_size(input_len, &config).unwrap();
+
+        assert_eq!(encoded_size, encode_config_slice(&orig_data, config, &mut encoded_data));
+
+        assert_encode_sanity(std::str::from_utf8(&encoded_data[0..encoded_size]).unwrap(),
+                             &config, input_len);
+
+        assert_eq!(&encoded_data[encoded_size..], &encoded_data_original_state[encoded_size..]);
+
+        // since we know we have the correct count of line endings, it's reasonable to simply remove
+        // them without worrying about where they are
+        let encoded_no_line_endings: String = String::from_utf8(encoded_data[0..encoded_size].iter()
+            .filter(|&b| *b != 0xA_u8 && *b != 0xD_u8)
```

Comment:
these could be chars instead of bytes, right? https://play.rust-lang.org/?gist=5c5e84b31b8dc90037fb9e03e0ae6b99

Reply:
Ah, good call. I think it's slightly more readable to cast from `char` to `u8`, which also seems to work well. Good thing those are all single-byte chars in UTF8.